contracts/cw3-flex-multisig/src/contract.rs

```diff
@@ -990,4 +1037,125 @@ mod tests {
             .unwrap_err();
         assert_eq!(err, ContractError::WrongCloseStatus {}.to_string());
     }
+
+    // Currently this just closes all open proposals
+    // TODO: something more clever (lazily tracking voting power at start of election)
+    #[test]
+    fn handle_group_changes() {
+        let mut app = mock_app();
+
+        let required_weight = 4;
+        let voting_period = Duration::Time(20000);
+        let (flex_addr, group_addr) = setup_test_case(
+            &mut app,
+            required_weight,
+            voting_period,
+            coins(10, "BTC"),
+            true,
+        );
+
+        // Start a proposal to remove VOTER3 from the set
+        let update_msg = Cw4Contract(group_addr.clone())
+            .update_members(vec![VOTER3.into()], vec![])
+            .unwrap();
+        let update_proposal = HandleMsg::Propose {
+            title: "Kick out VOTER3".to_string(),
+            description: "He's trying to steal our money".to_string(),
+            msgs: vec![update_msg],
+            latest: None,
+        };
+        let res = app
+            .execute_contract(VOTER1, &flex_addr, &update_proposal, &[])
```

Comment:
Nice, using the multisig to update group memberships. But, this is unnecessarily complex. I would leave this, as it's a nice test, but write another (simpler) test (or add a TODO for now) before this one, that just does the basic thing:

- Creates a group and associated multisig.
- Install the hook.
- Check that the multisig is Open.
- Change group memership, directly against the group addr.
- Check that the multisig is now Rejected / Aborted.